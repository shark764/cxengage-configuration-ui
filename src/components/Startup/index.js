import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { LoadingSpinnerSVG } from 'cx-ui-components';

const Loading = styled(LoadingSpinnerSVG)`
  position: absolute;
  top: calc(50% - ${props => props.size / 2}px);
  left: calc(50% - ${props => props.size / 2}px);
`;

export default class Startup extends Component {
  componentDidMount() {
    this.props.start();
  }

  render() {
    return this.props.hasStarted ? this.props.children : <Loading size={120} />;
  }
}

Startup.propTypes = {
  start: PropTypes.func.isRequired,
  hasStarted: PropTypes.bool.isRequired,
  children: PropTypes.any
};
