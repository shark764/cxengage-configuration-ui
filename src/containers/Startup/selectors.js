/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import {
  getBranding,
  getProtectedBranding
} from '../../redux/modules/entities/branding/selectors';

export const hasStarted = state =>
  getBranding(state) !== undefined && getProtectedBranding(state) !== undefined;
