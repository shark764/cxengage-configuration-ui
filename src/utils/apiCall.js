import { sdkPromise } from './sdk';

export function apiCall(urlSuffix, method = 'GET', data) {
  return sdkPromise({
    module: 'updateLocalStorage',
    command: `updateLocalStorage`,
    data: 'updateLocalStorage',
    topic: 'updateLocalStorage'
  }).then(({tenant: {tenantId}, token, baseUrl}) => {
    const url = `${baseUrl}/v1/tenants/${tenantId}/${urlSuffix}`;
    const requestParams =  {
      mode: "cors", // no-cors, cors, *same-origin
      cache: "no-cache", // *default, no-cache, reload, force-cache, only-if-cached
      Authorization: 'Token ' + token,
      headers: {
          "Content-Type": "application/json",
      },
      redirect: "follow", // manual, *follow, error
      referrer: "no-referrer", // no-referrer, *client
    };
    requestParams.method = method;
    if(token) {
      requestParams.headers.Authorization = 'Token ' + token;
    }
    if(data) {
      requestParams.body = JSON.stringify(data);
    }
      return fetch(url, requestParams)
      .then(response => {
        
        if(response.status === 401) {
          return 'Unauthorized';
        } else if ( response.status >= 300) {
          return;
        } else {
          return response.json();
        }
      });

  });
}

export function updateBetaFeatures(featuresObject) {
  return apiCall('settings/betaFeatures/value', 'PUT', featuresObject);
}

export function readBetaFeatures(featuresObject) {
  return apiCall('settings/betaFeatures/value');
}