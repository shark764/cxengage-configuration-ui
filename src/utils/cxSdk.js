export function cxInit() {
  return CxEngage.initialize(require('../sdkConfig.json'));
}

export function cxLogin(username, password, callback) {
  CxEngage.authentication.login({ 
    username: username, 
    password: password,
    // ttl is good to comment out to be able to test what 
    // happens when authentication expires...
    // ttl in seconds 
    // ttl: 60 
  }, (err, topic, response) => {
    if (err) {
      console.warn('ERROR WITH LOGIN', err);
      return callback(err);
    } else {
      return callback(response);
    }
  });
}

export function cxTokenLogin(token, callback) {
  CxEngage.authentication.login({ token: token }, (err, topic, response) => {
    if (err) {
      console.warn('ERROR WITH LOGIN', err);
      return callback(err);
    } else {
      return callback(response);
    }
  });
}

export function cxSetTenant(selectedTenant, callback) {
  CxEngage.session.setActiveTenant({ tenantId: selectedTenant, noSession: true }, (err, topic, response) => {
    if (err) {
      console.warn('ERROR WITH TENANT SELECTION', err);
    } else {
      return callback(response);
    }
  });
}
