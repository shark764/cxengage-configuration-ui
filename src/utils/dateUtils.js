export const dateToString = date =>
  new Date(date).toLocaleString('en-US', {
    month: 'short',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    second: '2-digit',
    hour12: true
  });
