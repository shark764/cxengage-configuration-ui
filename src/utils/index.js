export const detailHeaderText = (itemArray, name) => {
  if (itemArray) {
    return `${itemArray.length > 1 ? itemArray.length : ''} ${name}`;
  } else {
    return name;
  }
};

export const parentUrl = document.referrer;

const cloneObject = obj => Object.assign({}, obj);

export const renameObjectKey = (object, key, newKey) => {
  const clonedObj = cloneObject(object);
  const targetKey = clonedObj[key];
  delete clonedObj[key];
  clonedObj[newKey] = targetKey;
  return clonedObj;
};

// Sort method to sort strings in entity tables
export const customSortStringMethod = (a, b) => {
  let tempArray = [];
  tempArray = [a, b];

  const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'base' });
  tempArray.sort(collator.compare);

  if (tempArray[0] === a) {
    return -1;
  } else {
    return 1;
  }
};