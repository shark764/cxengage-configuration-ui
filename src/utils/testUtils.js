/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { createStore } from 'redux';

export const mockStore = createStore(state => state);
