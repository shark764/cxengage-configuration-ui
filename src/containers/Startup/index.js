/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { connect } from 'react-redux';

import Startup from '../../components/Startup';

import { hasStarted } from './selectors';
import { fetchBranding } from '../../redux/modules/entities/branding/actions';

export function mapStateToProps(state) {
  return {
    hasStarted: hasStarted(state)
  };
}

export const actions = {
  start: fetchBranding
};

const StartupContainer = connect(mapStateToProps, actions)(Startup);

export default StartupContainer;
