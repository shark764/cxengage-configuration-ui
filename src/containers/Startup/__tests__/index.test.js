/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { mockStore } from '../../../utils/testUtils';
import Startup, { mapStateToProps, actions } from '../';

jest.mock('../selectors', () => ({
  hasStarted: () => true
}));

jest.mock('../../../redux/modules/entities/branding/actions');

describe('Startup', () => {
  it('renders', () => {
    shallow(<Startup store={mockStore} />);
  });
});

describe('mapStateToProps', () => {
  it('maps the selectors to the object correctly', () => {
    expect(mapStateToProps()).toMatchSnapshot();
  });
});

describe('actions', () => {
  it('maps the redux actions to the object correctly', () => {
    expect(actions).toMatchSnapshot();
  });
});
