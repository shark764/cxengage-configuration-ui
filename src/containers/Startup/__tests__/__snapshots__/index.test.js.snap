// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions maps the redux actions to the object correctly 1`] = `
Object {
  "start": [MockFunction],
}
`;

exports[`mapStateToProps maps the selectors to the object correctly 1`] = `
Object {
  "hasStarted": true,
}
`;
